bitwarden_license/bit-web/src/app/sm/secrets/secret.service.ts

```diff
@@ -0,0 +1,122 @@
+    const r = await this.apiService.send("GET", "/secrets/" + secretId, null, true, true);
+    const secretResponse = new SecretResponse(r);
+    return await this.decryptSecretView(secretResponse.toSecretView());
+  }
+
+  async getSecrets(organizationId: string): Promise<SecretListView[]> {
+    const r = await this.apiService.send(
+      "GET",
+      "/organizations/" + organizationId + "/secrets",
+      null,
+      true,
+      true
+    );
+    const results = new ListResponse(r, SecretIdentifierResponse);
+    const secrets: SecretListView[] = results.data.map(function (result) {
+      return result.toSecretListView();
+    });
+    return await this.decryptSecretsListView(organizationId, secrets);
+  }
+
+  async create(organizationId: string, secretView: SecretView) {
+    const encryptedSecretView = await this.encryptSecretView(organizationId, secretView);
+
+    const request = new CreateSecretRequest();
+    request.organizationId = organizationId;
+    request.key = encryptedSecretView.name;
+    request.value = encryptedSecretView.value;
+    request.note = encryptedSecretView.note;
+
+    const r = await this.apiService.send(
+      "POST",
+      "/organizations/" + organizationId + "/secrets",
+      request,
+      true,
+      true
+    );
+    this._secret.next(new SecretResponse(r).toSecretView());
+  }
+
+  async update(organizationId: string, secretView: SecretView) {
+    const encryptedSecretView = await this.encryptSecretView(organizationId, secretView);
+
+    const request = new UpdateSecretRequest();
+    request.key = encryptedSecretView.name;
+    request.value = encryptedSecretView.value;
+    request.note = encryptedSecretView.note;
+
+    const r = await this.apiService.send("PUT", "/secrets/" + secretView.id, request, true, true);
+    this._secret.next(new SecretResponse(r).toSecretView());
+  }
+
+  private async getOrganizationKey(organizationId: string) {
+    return await this.cryptoService.getOrgKey(organizationId);
+  }
+
+  private async decryptSecretsListView(organizationId: string, secrets: SecretListView[]) {
+    const orgKey: SymmetricCryptoKey = await this.getOrganizationKey(organizationId);
+    secrets.forEach(async (secrets) => {
+      secrets.name = await this.encryptService.decryptToUtf8(new EncString(secrets.name), orgKey);
+    });
+    return secrets;
+  }
+
+  private async encryptSecretView(organizationId: string, secretView: SecretView) {
+    const orgKey: SymmetricCryptoKey = await this.getOrganizationKey(organizationId);
+    secretView.name = (await this.encryptService.encrypt(secretView.name, orgKey)).encryptedString;
+    secretView.value = (
+      await this.encryptService.encrypt(secretView.value, orgKey)
+    ).encryptedString;
+    secretView.note = (await this.encryptService.encrypt(secretView.note, orgKey)).encryptedString;
+    return secretView;
+  }
+
+  private async decryptSecretView(secretView: SecretView) {
```

Comment:
We shouldn't store encrypted values in `SecretView`, lets change the interface to better represent what's going on here.

```suggestion
  private async decryptSecret(secret: Secret): SecretView {
```

Reply:
resolved in https://github.com/bitwarden/clients/pull/3376/commits/547717ca85458139e2b099d4eb974cacfd8a55b2